README.md
freenas-bacula
==============

Bacula File Deamon plugin for FreeNAS.

Forked from williambr/freenas-bacula

Adding some additional features including...

  Timeout value selection for the File Deamon.


CHANGELOG
1.0-BETA1
======================
	- Updated to FreeNAS 9.x conformance
	- Added the "Heartbeat Interval" options
	- Imported bug fixes and enhancements

0.1-BETA2
======================

    - Add initial migration
    - Fix service start/stop, there was some logic missing
    - Write rc.conf on bacula start/stop
    - Add delete buttons
    - Fix device assignment edit screen
    - Workaround install environment bug in 8.2.0

0.1-BETA1
======================

    - First public release of the plugin for FreeNAS 8.2


nanobsd/plugins/bacula_pbi/resources/baculaUI/freenas/models.py
from django.db import models
from django.utils.translation import ugettext_lazy as _
from django.core.validators import MinValueValidator, MaxValueValidator

from baculaUI import choices


class BaculaSDService(models.Model):
    enable = models.BooleanField(default=False)

class BaculaSDStorage(models.Model):
    """
    Django model describing every tunable setting for bacula
    """

    baculasd_st_name = models.CharField(
            default="freenas-sd",
            help_text=_("Specifies the Name of the Storage daemon. This "
                "directive is required."),
            max_length=255,
            unique=True,
            verbose_name=_("Name"),
            )
    baculasd_st_sdport = models.IntegerField(
            default=9103,
            help_text=_("Port number on which the Storage daemon listens for "
                "Director connections"),
            unique=True,
            validators=[MinValueValidator(1), MaxValueValidator(65535)],
            verbose_name=_("Port"),
            )
    baculasd_st_maximumconcurrentjobs = models.IntegerField(
            default=10,
            help_text=_("Maximum number of Jobs that may run concurrently."),
            verbose_name=_("Maximum Concurrent Jobs"),
            )
    baculasd_st_heartbeatinterval = models.IntegerField(
            default=0,
            help_text=_("Specifies a Heartbeat (keepalive) Interval in seconds."),
            verbose_name=_("Heartbeat Interval"),
            )
    baculasd_st_proceeddespiteioerrors = models.BooleanField(
            default=False,
            verbose_name=_("Proceed despite I/O errors"),
            )

    def __unicode__(self):
        return self.baculasd_st_name

    class Meta:
        verbose_name = _("Storage Daemon")
        verbose_name_plural = _("Storage Daemons")


class BaculaSDDirector(models.Model):
    baculasd_dir_name = models.CharField(
            default="bacula-dir",
            help_text=_("Specifies the Name of the Director allowed to connect"
                "  to the Storage daemon. This directive is required."),
            max_length=255,
            unique=True,
            verbose_name=_("Name"),
            )
    baculasd_dir_password = models.CharField(
            default="",
            help_text=_("Specifies the password that must be supplied by the "
                "above named Director. This directive is required."),
            max_length=255,
            verbose_name=_("Password"),
            )
    baculasd_dir_monitor = models.BooleanField(
            default=False,
            help_text=_("If Monitor is set to no (default), this director "
                "will have full access to this Storage daemon. If Monitor is "
                "set to yes, this director will only be able to fetch the "
                "current status of this Storage daemon."),
            verbose_name=_("Monitor"),
            )

    def __unicode__(self):
        return self.baculasd_dir_name

    class Meta:
        verbose_name = _("Director")
        verbose_name_plural = _("Directors")


class BaculaSDDevice(models.Model):
    baculasd_dev_name = models.CharField(
            default="FileStorage",
            help_text=_("Specifies the Name that the Director will use when "
                "asking to backup or restore to or from to this device."),
            max_length=127,
            unique=True,
            verbose_name=_("Name"),
            )
    baculasd_dev_mediatype = models.CharField(
            choices=choices.BACULASD_DEV_MEDIATYPE_CHOICES,
            default="File",
            help_text=_("The Media Type specification allows you to explicitly"
                " tell Bacula what kind of device you are defining."),
            max_length=5,
            verbose_name=_("Media Type"),
            )
    baculasd_dev_archivedevice = models.CharField(
            default="/mnt/tank/bacula",
            help_text=_("The specified name-string gives the system file name"
                " of the storage device managed by this storage daemon."),
            max_length=255,
            unique=True,
            verbose_name=_("Archive Device"),
            )
    baculasd_dev_labelmedia = models.BooleanField(
            default=True,
            help_text=_("If Yes, permits this device to automatically label "
                "blank media without an explicit operator command."),
            verbose_name=_("Label Media"),
            )
    baculasd_dev_randomaccess = models.BooleanField(
            default=True,
            help_text=_("If Yes, the archive device is assumed to be a random"
                " access medium which supports the lseek (or lseek64 if "
                "Largefile is enabled during configuration) facility."),
            verbose_name=_("Random access"),
            )
    baculasd_dev_removablemedia = models.BooleanField(
            default=False,
            help_text=_("If Yes, this device supports removable media (for "
                "example, tapes or CDs). If No, media cannot be removed (for "
                "example, an intermediate backup area on a hard disk)."),
            verbose_name=_("Removable media"),
            )
    baculasd_dev_alwaysopen = models.BooleanField(
            default=False,
            help_text=_("If Yes (default), Bacula will always keep the device"
                "open unless specifically unmounted by the Console program."),
            verbose_name=_("Always open"),
            )
    baculasd_dev_maximumconcurrentjobs = models.IntegerField(
            blank=True,
            default=0,
            help_text=_("Maximum number of Jobs that can run concurrently on "
                "this Device."),
            verbose_name=_("Maximum Concurrent Jobs"),
            )

    class Meta:
        verbose_name = _("Device")
        verbose_name_plural = _("Devices")

    def __unicode__(self):
        return unicode(self.baculasd_dev_name)


class BaculaSDMessages(models.Model):
    baculasd_msg_name = models.CharField(
            default="Standard",
            help_text=_("The name of the Messages resource."),
            max_length=255,
            unique=True,
            verbose_name=_("Name"),
            )
    baculasd_msg_destination = models.CharField(
            choices=choices.BACULASD_MSG_DESTINATION_CHOICES,
            default="director",
            help_text=_("The destinations to which messages should be sent."),
            max_length=15,
            verbose_name=_("Destination"),
            )
    baculasd_msg_address = models.CharField(
            blank=True,
            default="bacula-dir",
            help_text=_("The destinations address."),
            max_length=255,
            verbose_name=_("Address"),
            )
    baculasd_msg_msgtype1 = models.CharField(
            choices=choices.BACULASD_MSG_MSGTYPE_CHOICES,
            default="all",
            max_length=9,
            verbose_name=_("Message Type 1"),
            )
    baculasd_msg_msgtype2 = models.CharField(
            blank=True,
            choices=choices.BACULASD_MSG_MSGTYPE_CHOICES,
            default="",
            max_length=9,
            verbose_name=_("Message Type 2"),
            )
    baculasd_msg_msgtype3 = models.CharField(
            blank=True,
            choices=choices.BACULASD_MSG_MSGTYPE_CHOICES,
            default="",
            max_length=9,
            verbose_name=_("Message Type 3"),
            )
    baculasd_msg_msgtypeaux = models.TextField(
            blank=True,
            default="# message-type4, message-type5, message-type6, ... ",
            help_text=_("Comma separated list of additional message types."),
            verbose_name=_("Auxiliary message types"),
            )

    class Meta:
        verbose_name = _("Messages")
        verbose_name_plural = _("Messages")

    def __unicode__(self):
        return unicode(self.baculasd_msg_name)


class BaculaSDDeviceAssignment(models.Model):
    baculasd_map_device = models.ForeignKey(
            BaculaSDDevice,
            verbose_name=_("Device resource"),
            )
    baculasd_map_storage = models.ForeignKey(
            BaculaSDStorage,
            verbose_name=_("Storage Daemon"),
            )

    class Meta:
        verbose_name = _("Device assignment")
        verbose_name_plural = _("Device assignments")

    def __unicode__(self):
        return u"%s -> %s" % (
            unicode(self.baculasd_map_device),
            unicode(self.baculasd_map_storage),
            )


class BaculaSDDirectorAssignment(models.Model):
    baculasd_map_director = models.ForeignKey(
            BaculaSDDirector,
            verbose_name=_("Director resource"),
            )
    baculasd_map_storage = models.ForeignKey(
            BaculaSDStorage,
            verbose_name=_("Storage Daemon"),
            )

    class Meta:
        verbose_name = _("Director assignment")
        verbose_name_plural = _("Director assignments")

    class FreeAdmin:
        icon_model = u"BaculaDirectorMapIcon"
        icon_object = u"BaculaDirectorMapIcon"
        menu_child_of = "BaculaSD"

    def __unicode__(self):
        return u"%s -> %s" % (
            unicode(self.baculasd_map_director),
            unicode(self.baculasd_map_storage),
            )


class BaculaSDMessagesAssignment(models.Model):
    baculasd_map_messages = models.ForeignKey(
            BaculaSDMessages,
            verbose_name=_("Messages resource"),
            )
    baculasd_map_storage = models.ForeignKey(
            BaculaSDStorage,
            verbose_name=_("Storage Daemon"),
            )

    class Meta:
        verbose_name = _("Messages assignment")
        verbose_name_plural = _("Messages assignments")

    def __unicode__(self):
        return u"%s -> %s" % (
            unicode(self.baculasd_map_messages),
            unicode(self.baculasd_map_storage),
            )


nanobsd/plugins/bacula_pbi/resources/baculaUI/freenas/migrations/0002_heartbeat_interval.py
# -*- coding: utf-8 -*-
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


class Migration(SchemaMigration):
    def forwards(self, orm):
        # Adding the heartbeatinterval column
            db.add_column('freenas_baculasdstorage', 'baculasd_st_heartbeatinterval',
                          self.gf('django.db.models.fields.IntegerField')(default=0))

    def backwards(self, orm):
        # Delete the heartbeatinterval column
            db.delete_column('freenas_baculasdstorage', 'baculasd_st_heartbeatinterval')

    models = {
        'freenas.baculasddevice': {
            'Meta': {'object_name': 'BaculaSDDevice'},
            'baculasd_dev_alwaysopen': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_dev_archivedevice': ('django.db.models.fields.CharField', [], {'default': "'/mnt/tank/bacula'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_dev_labelmedia': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'baculasd_dev_maximumconcurrentjobs': ('django.db.models.fields.IntegerField', [], {'default': '0', 'blank': 'True'}),
            'baculasd_dev_mediatype': ('django.db.models.fields.CharField', [], {'default': "'File'", 'max_length': '5'}),
            'baculasd_dev_name': ('django.db.models.fields.CharField', [], {'default': "'FileStorage'", 'unique': 'True', 'max_length': '127'}),
            'baculasd_dev_randomaccess': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'baculasd_dev_removablemedia': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddeviceassignment': {
            'Meta': {'object_name': 'BaculaSDDeviceAssignment'},
            'baculasd_map_device': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDDevice']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddirector': {
            'Meta': {'object_name': 'BaculaSDDirector'},
            'baculasd_dir_monitor': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_dir_name': ('django.db.models.fields.CharField', [], {'default': "'bacula-dir'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_dir_password': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '255'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddirectorassignment': {
            'Meta': {'object_name': 'BaculaSDDirectorAssignment'},
            'baculasd_map_director': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDDirector']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdmessages': {
            'Meta': {'object_name': 'BaculaSDMessages'},
            'baculasd_msg_address': ('django.db.models.fields.CharField', [], {'default': "'bacula-dir'", 'max_length': '255', 'blank': 'True'}),
            'baculasd_msg_destination': ('django.db.models.fields.CharField', [], {'default': "'director'", 'max_length': '15'}),
            'baculasd_msg_msgtype1': ('django.db.models.fields.CharField', [], {'default': "'all'", 'max_length': '9'}),
            'baculasd_msg_msgtype2': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '9', 'blank': 'True'}),
            'baculasd_msg_msgtype3': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '9', 'blank': 'True'}),
            'baculasd_msg_msgtypeaux': ('django.db.models.fields.TextField', [], {'default': "'# message-type4, message-type5, message-type6, ... '", 'blank': 'True'}),
            'baculasd_msg_name': ('django.db.models.fields.CharField', [], {'default': "'Standard'", 'unique': 'True', 'max_length': '255'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdmessagesassignment': {
            'Meta': {'object_name': 'BaculaSDMessagesAssignment'},
            'baculasd_map_messages': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDMessages']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdservice': {
            'Meta': {'object_name': 'BaculaSDService'},
            'enable': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdstorage': {
            'Meta': {'object_name': 'BaculaSDStorage'},
            'baculasd_st_maximumconcurrentjobs': ('django.db.models.fields.IntegerField', [], {'default': '10'}),
            'baculasd_st_heartbeatinterval': ('django.db.models.fields.IntegerField', [], {'default': '0'}),
            'baculasd_st_name': ('django.db.models.fields.CharField', [], {'default': "'freenas-sd'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_st_proceeddespiteioerrors': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_st_sdport': ('django.db.models.fields.IntegerField', [], {'default': '9103', 'unique': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        }
    }

    complete_apps = ['freenas']


nanobsd/plugins/bacula_pbi/resources/baculaUI/freenas/migrations/0001_initial.py
# -*- coding: utf-8 -*-
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


class Migration(SchemaMigration):

    def forwards(self, orm):
        # Adding model 'BaculaSDService'
        db.create_table('freenas_baculasdservice', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('enable', self.gf('django.db.models.fields.BooleanField')(default=False)),
        ))
        db.send_create_signal('freenas', ['BaculaSDService'])

        # Adding model 'BaculaSDStorage'
        db.create_table('freenas_baculasdstorage', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_st_name', self.gf('django.db.models.fields.CharField')(default='freenas-sd', unique=True, max_length=255)),
            ('baculasd_st_sdport', self.gf('django.db.models.fields.IntegerField')(default=9103, unique=True)),
            ('baculasd_st_maximumconcurrentjobs', self.gf('django.db.models.fields.IntegerField')(default=10)),
            ('baculasd_st_proceeddespiteioerrors', self.gf('django.db.models.fields.BooleanField')(default=False)),
        ))
        db.send_create_signal('freenas', ['BaculaSDStorage'])

        # Adding model 'BaculaSDDirector'
        db.create_table('freenas_baculasddirector', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_dir_name', self.gf('django.db.models.fields.CharField')(default='bacula-dir', unique=True, max_length=255)),
            ('baculasd_dir_password', self.gf('django.db.models.fields.CharField')(default='', max_length=255)),
            ('baculasd_dir_monitor', self.gf('django.db.models.fields.BooleanField')(default=False)),
        ))
        db.send_create_signal('freenas', ['BaculaSDDirector'])

        # Adding model 'BaculaSDDevice'
        db.create_table('freenas_baculasddevice', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_dev_name', self.gf('django.db.models.fields.CharField')(default='FileStorage', unique=True, max_length=127)),
            ('baculasd_dev_mediatype', self.gf('django.db.models.fields.CharField')(default='File', max_length=5)),
            ('baculasd_dev_archivedevice', self.gf('django.db.models.fields.CharField')(default='/mnt/tank/bacula', unique=True, max_length=255)),
            ('baculasd_dev_labelmedia', self.gf('django.db.models.fields.BooleanField')(default=True)),
            ('baculasd_dev_randomaccess', self.gf('django.db.models.fields.BooleanField')(default=True)),
            ('baculasd_dev_removablemedia', self.gf('django.db.models.fields.BooleanField')(default=False)),
            ('baculasd_dev_alwaysopen', self.gf('django.db.models.fields.BooleanField')(default=False)),
            ('baculasd_dev_maximumconcurrentjobs', self.gf('django.db.models.fields.IntegerField')(default=0, blank=True)),
        ))
        db.send_create_signal('freenas', ['BaculaSDDevice'])

        # Adding model 'BaculaSDMessages'
        db.create_table('freenas_baculasdmessages', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_msg_name', self.gf('django.db.models.fields.CharField')(default='Standard', unique=True, max_length=255)),
            ('baculasd_msg_destination', self.gf('django.db.models.fields.CharField')(default='director', max_length=15)),
            ('baculasd_msg_address', self.gf('django.db.models.fields.CharField')(default='bacula-dir', max_length=255, blank=True)),
            ('baculasd_msg_msgtype1', self.gf('django.db.models.fields.CharField')(default='all', max_length=9)),
            ('baculasd_msg_msgtype2', self.gf('django.db.models.fields.CharField')(default='', max_length=9, blank=True)),
            ('baculasd_msg_msgtype3', self.gf('django.db.models.fields.CharField')(default='', max_length=9, blank=True)),
            ('baculasd_msg_msgtypeaux', self.gf('django.db.models.fields.TextField')(default='# message-type4, message-type5, message-type6, ... ', blank=True)),
        ))
        db.send_create_signal('freenas', ['BaculaSDMessages'])

        # Adding model 'BaculaSDDeviceAssignment'
        db.create_table('freenas_baculasddeviceassignment', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_map_device', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDDevice'])),
            ('baculasd_map_storage', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDStorage'])),
        ))
        db.send_create_signal('freenas', ['BaculaSDDeviceAssignment'])

        # Adding model 'BaculaSDDirectorAssignment'
        db.create_table('freenas_baculasddirectorassignment', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_map_director', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDDirector'])),
            ('baculasd_map_storage', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDStorage'])),
        ))
        db.send_create_signal('freenas', ['BaculaSDDirectorAssignment'])

        # Adding model 'BaculaSDMessagesAssignment'
        db.create_table('freenas_baculasdmessagesassignment', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('baculasd_map_messages', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDMessages'])),
            ('baculasd_map_storage', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['freenas.BaculaSDStorage'])),
        ))
        db.send_create_signal('freenas', ['BaculaSDMessagesAssignment'])


    def backwards(self, orm):
        # Deleting model 'BaculaSDService'
        db.delete_table('freenas_baculasdservice')

        # Deleting model 'BaculaSDStorage'
        db.delete_table('freenas_baculasdstorage')

        # Deleting model 'BaculaSDDirector'
        db.delete_table('freenas_baculasddirector')

        # Deleting model 'BaculaSDDevice'
        db.delete_table('freenas_baculasddevice')

        # Deleting model 'BaculaSDMessages'
        db.delete_table('freenas_baculasdmessages')

        # Deleting model 'BaculaSDDeviceAssignment'
        db.delete_table('freenas_baculasddeviceassignment')

        # Deleting model 'BaculaSDDirectorAssignment'
        db.delete_table('freenas_baculasddirectorassignment')

        # Deleting model 'BaculaSDMessagesAssignment'
        db.delete_table('freenas_baculasdmessagesassignment')


    models = {
        'freenas.baculasddevice': {
            'Meta': {'object_name': 'BaculaSDDevice'},
            'baculasd_dev_alwaysopen': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_dev_archivedevice': ('django.db.models.fields.CharField', [], {'default': "'/mnt/tank/bacula'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_dev_labelmedia': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'baculasd_dev_maximumconcurrentjobs': ('django.db.models.fields.IntegerField', [], {'default': '0', 'blank': 'True'}),
            'baculasd_dev_mediatype': ('django.db.models.fields.CharField', [], {'default': "'File'", 'max_length': '5'}),
            'baculasd_dev_name': ('django.db.models.fields.CharField', [], {'default': "'FileStorage'", 'unique': 'True', 'max_length': '127'}),
            'baculasd_dev_randomaccess': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'baculasd_dev_removablemedia': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddeviceassignment': {
            'Meta': {'object_name': 'BaculaSDDeviceAssignment'},
            'baculasd_map_device': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDDevice']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddirector': {
            'Meta': {'object_name': 'BaculaSDDirector'},
            'baculasd_dir_monitor': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_dir_name': ('django.db.models.fields.CharField', [], {'default': "'bacula-dir'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_dir_password': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '255'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasddirectorassignment': {
            'Meta': {'object_name': 'BaculaSDDirectorAssignment'},
            'baculasd_map_director': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDDirector']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdmessages': {
            'Meta': {'object_name': 'BaculaSDMessages'},
            'baculasd_msg_address': ('django.db.models.fields.CharField', [], {'default': "'bacula-dir'", 'max_length': '255', 'blank': 'True'}),
            'baculasd_msg_destination': ('django.db.models.fields.CharField', [], {'default': "'director'", 'max_length': '15'}),
            'baculasd_msg_msgtype1': ('django.db.models.fields.CharField', [], {'default': "'all'", 'max_length': '9'}),
            'baculasd_msg_msgtype2': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '9', 'blank': 'True'}),
            'baculasd_msg_msgtype3': ('django.db.models.fields.CharField', [], {'default': "''", 'max_length': '9', 'blank': 'True'}),
            'baculasd_msg_msgtypeaux': ('django.db.models.fields.TextField', [], {'default': "'# message-type4, message-type5, message-type6, ... '", 'blank': 'True'}),
            'baculasd_msg_name': ('django.db.models.fields.CharField', [], {'default': "'Standard'", 'unique': 'True', 'max_length': '255'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdmessagesassignment': {
            'Meta': {'object_name': 'BaculaSDMessagesAssignment'},
            'baculasd_map_messages': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDMessages']"}),
            'baculasd_map_storage': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['freenas.BaculaSDStorage']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdservice': {
            'Meta': {'object_name': 'BaculaSDService'},
            'enable': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        },
        'freenas.baculasdstorage': {
            'Meta': {'object_name': 'BaculaSDStorage'},
            'baculasd_st_maximumconcurrentjobs': ('django.db.models.fields.IntegerField', [], {'default': '10'}),
            'baculasd_st_name': ('django.db.models.fields.CharField', [], {'default': "'freenas-sd'", 'unique': 'True', 'max_length': '255'}),
            'baculasd_st_proceeddespiteioerrors': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'baculasd_st_sdport': ('django.db.models.fields.IntegerField', [], {'default': '9103', 'unique': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'})
        }
    }

    complete_apps = ['freenas']
